### PR TITLE
fix: close modal if no need to reauth

### PR DESCRIPTION
**Pull request by tomiir**
# Summary
Fix issue where an integration with `signOutOnNetworkChange: false` would get stuck on the loading modal after switching in the wallet. 

# Changes

- feat:
- fix: close modal if there's no need to re-auth SIWE
- chore:

# Associated Issues

closes #APKT-596

### CODE REVIEW

File: packages/scaffold/src/views/w3m-network-switch-view/index.ts [hunk 1]
```diff
@@ -41,6 +39,7 @@ export class W3mNetworkSwitchView extends LitElement {
 
   // -- Render -------------------------------------------- //
   public override render() {
+    console.log('SWITCH NETWORK VIEW')
```
Comment (enesozturk) [inline packages/scaffold/src/views/w3m-network-switch-view/index.ts:42]: Log

File: packages/scaffold/src/modal/w3m-modal/index.ts [hunk 2]
```diff
@@ -189,16 +187,6 @@ export class W3mModal extends LitElement {
         return
       }
 
-      // If the network has changed and signOnNetworkChange is enabled, sign out
```
Comment (enesozturk) [inline packages/scaffold/src/modal/w3m-modal/index.ts:192]: What if the user change the network from the wallet when the w3m is closed state? I guess this is the only place that will trigger SIWE 🤔

File: apps/laboratory/tests/siwe.spec.ts [hunk 3]
```diff
@@ -32,3 +32,21 @@ testMWSiwe(
     await modalValidator.expectUnauthenticated()
   }
 )
+
+testMWSiwe(
```
Comment (Cali93) [inline apps/laboratory/tests/siwe.spec.ts:36]: Awesome to see a test covering this 💪

File: packages/scaffold/src/views/w3m-network-switch-view/index.ts [hunk 4]
```diff
@@ -134,7 +134,12 @@ export class W3mNetworkSwitchView extends LitElement {
       this.error = false
       if (this.network) {
         await NetworkController.switchActiveNetwork(this.network)
-        if (!OptionsController.state.isSiweEnabled) {
+        if (OptionsController.state.isSiweEnabled) {
+          const { SIWEController } = await import('@web3modal/siwe')
+          if (SIWEController.state._client?.options?.signOutOnNetworkChange) {
+            await SIWEController.signOut()
+          }
+        } else {
```
Comment (Cali93) [inline packages/scaffold/src/views/w3m-network-switch-view/index.ts:142]: Should this logic live in `switchActiveNetwork` to avoid code duplication ?

**Review by Cali93 — APPROVED** [review]
LGTM, just a minor suggestion to maybe abstract this logic to avoid duplication.

File: packages/scaffold/src/views/w3m-network-switch-view/index.ts (hunk 4 above)
Comment (tomiir) [reply packages/scaffold/src/views/w3m-network-switch-view/index.ts:142]: That causes a circular dependency cause it'd require `siwe` in `core` which is imported in `siwe` 😵 
But agree not cool to duplicate code

Comment (tomiir) [reply packages/scaffold/src/views/w3m-network-switch-view/index.ts]: abstracted to a utility 🙏

**Review by enesozturk — APPROVED** [review]

**Review by svenvoskamp — APPROVED** [review]

**Pull request merged**